modules/backend/models/ExportModel.php

```diff
@@ -53,16 +55,17 @@ public function export($columns, $options)
      */
     public function download($name, $outputName = null)
     {
-        if (!preg_match('/^oc[0-9a-z]*$/i', $name)) {
-            throw new ApplicationException(Lang::get('backend::lang.import_export.file_not_found_error'));
-        }
-
-        $csvPath = temp_path() . '/' . $name;
-        if (!file_exists($csvPath)) {
-            throw new ApplicationException(Lang::get('backend::lang.import_export.file_not_found_error'));
-        }
-
-        return Response::download($csvPath, $outputName)->deleteFileAfterSend(true);
+        $cachedExport = Cache::get('export::' . $name);
+        return new StreamedResponse(
+            function () use ($cachedExport) {
+                echo $cachedExport;
+            },
+            200,
+            [
+                'Content-Type' => 'text/csv',
+                'Content-Disposition' => 'attachment; filename="' . $outputName . '"',
```

Comment:
@Samuell1 You will need to add some handling to have a fallback value for `$outputName` if it is `null`, otherwise it will name the file as `null` when downloading, I believe. The old `Response::download()` call would use the filename of the source file if an `$outputName` was not specified.

Reply:
@bennothommo Old Response::download() is using export.csv as $outputName, because its automatically passed from download controller.

Reply:
@Samuell1 No problem, if that's the case, that works.